Composite/Core/Routing/Pages/C1PageRouteHander.cs

```diff
@@ -94,8 +101,28 @@ public IHttpHandler GetHttpHandler(RequestContext requestContext)
             {
                 return (IHttpHandler)Activator.CreateInstance(_handlerType);
             }
-            
-            return (IHttpHandler)BuildManager.CreateInstanceFromVirtualPath("~/Renderers/Page.aspx", typeof(Page));
+
+            if (GlobalSettingsFacade.OmitAspNetWebFormsSupport)
+            {
+                var page = _pageUrlData.GetPage()
+                    ?? throw new HttpException(404, "Page not found - either this page has not been published yet or it has been deleted.");
+
+                if (IsSlimPageRenderer(page.TemplateId))
+                {
+                    return new CmsPageHttpHandler();

```

Comment:
The cache used also for unpublished pages
![image](https://user-images.githubusercontent.com/14817424/72250949-7aba7d80-3605-11ea-8901-094f7e52a006.png)

Reply:
Fixed